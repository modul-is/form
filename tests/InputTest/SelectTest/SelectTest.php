<?php

declare(strict_types=1);

require_once __DIR__ . '/../../bootstrap.php';

use Tester\Assert;

class SelectTest extends Tester\TestCase
{
	public function testRender()
	{
		$form = new ModulIS\Form\Form;

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second']);

		$html = str_replace(["\t", "\n"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('select')->render()->__toString());
	}


	public function testRenderOptionId()
	{
		$form = new ModulIS\Form\Form;

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second'])
			->setOption('id', 'customId');

		$string = '<div class="form-group row" id="customId"><div class="col-sm-4 control-label align-self-center"><label for="frm-select" class="col-form-label ">Select</label></div><div class="col-sm-8"><div class="input-group"><select name="select" id="frm-select" class="form-control  "><option value="first">First</option><option value="second">Second</option></select></div></div></div>';

		Assert::same($string, $form->getComponent('select')->render()->__toString());
	}
	
	
	public function testRenderInputFloatingLabel()
	{
		$form = new ModulIS\Form\Form;

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second'])
			->setFloatingLabel(true);

		$html = str_replace(["\t", "\n"], '', file_get_contents(__DIR__ . '/floatingLabel.latte'));

		Assert::same($html, $form->getComponent('select')->render()->__toString());
	}
	
	
	public function testRenderFormFloatingLabel()
	{
		$form = new ModulIS\Form\Form;
		
		$form->setFloatingLabel(true);

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second']);

		$html = str_replace(["\t", "\n"], '', file_get_contents(__DIR__ . '/floatingLabel.latte'));

		Assert::same($html, $form->getComponent('select')->render()->__toString());
	}


	public function testRenderFormFloatingLabelInputDisable()
	{
		$form = new ModulIS\Form\Form;
		
		$form->setFloatingLabel(true);

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second'])
			->setFloatingLabel(false);

		$html = str_replace(["\t", "\n"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('select')->render()->__toString());
	}


	public function testRenderCustomTemplate()
	{
		$form = new ModulIS\Form\Form;

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second'])
			->setTemplate(__DIR__ . '/custom.latte');

		$string = 'custom-template';

		Assert::same($string, $form->getComponent('select')->render());
	}


	public function testRenderHidden()
	{
		$form = new ModulIS\Form\Form;

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second'])
			->setOption('hide', true);

		$string = '';

		Assert::same($string, $form->getComponent('select')->render());
	}


	public function testRenderSkip()
	{
		$form = new ModulIS\Form\Form;

		$form->addSelect('select', 'Select', ['first' => 'First', 'second' => 'Second'])
			->setAutoRenderSkip();

		$string = '';

		Assert::same($string, $form->getComponent('select')->render());
	}
}

$testcase = new SelectTest;
$testcase->run();
