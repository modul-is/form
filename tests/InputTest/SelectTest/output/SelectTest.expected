<div class="mb-3 col-12"><div class="row"><div class="align-self-center col-sm-4"><label for="frm-form-select" class="col-form-label">Select</label></div><div class="align-self-center col-sm-8"><select name="select" id="frm-form-select" style="display: none;"><option value="">~ Vyberte ~</option><option value="first">First</option><option value="second">Second</option></select><div id="frm-form-select-wrapper" data-parent-id="frm-form-select" class="dropdown select-image d-block"><a class="btn dropdown-toggle d-block" id="frm-form-select-dropdown" data-bs-toggle="dropdown" aria-expanded="false" type="button"><span class="prompt-text float-start">~ Vyberte ~</span></a><ul aria-labelledby="frm-form-select-dropdown" class="dropdown-menu ps-2 pe-2 w-100"><li value=""><a class="dropdown-item"><div class="label-text mt-1">~ Vyberte ~</div></a></li><li value="first"><a class="row dropdown-item"><div class="image-wrap"><img src="/images/first.png"></div><div class="label-text mt-1 ms-2">First</div></a></li><li value="second"><a class="row dropdown-item"><div class="image-wrap"><img src="/images/second.png"></div><div class="label-text mt-1 ms-2">Second</div></a></li></ul></div></div></div></div>